doc/CCLUTMOCKING.md

```diff
@@ -212,6 +212,17 @@ call cclutRemoveAllMockImplementations(null)
 7. The table mocking APIs are not supported when called from within a reportwriter section.  It might be tempting to use a dummyt query to set up mock data from a record structure, but various mocking calls such as cclutCreateMockTable, cclutRemoveMockTable and cclutAddMockData cannot be executed within the context of a query (because the implementations execute queries). Use a for loop instead.
  
 8. Mocking the tdbexecute "reply_to" entity is unsupported under certain conditions, specifically if a call to free the "reply_to" entity is made just prior to calling tdbexecute.  If the scenario is truly necessary for a test, the best alternative is to define and use a subroutine for freeing the "reply_to" entity within the script and use a mock for that subroutine which does not actually perform the freeing of the "reply_to" entity.
+
+9. In versions of CCL prior to 8.14.4 (32-bit) and 9.2.4 (64-bit), mocking is not supported in situations where all of the following are true:
+    - The item to be mocked is inside of a subroutine declared using the "subroutine" syntax instead of the "declare" syntax.
+    - The item to be mocked is inside of a subroutine declared with the "with copy" option.
+    - A program is executed in the script-under-test before the first invocation of the subroutine containing the item to be mocked.
+
+   The simplest way to avoid this if all three are true is to change to the older "declare" syntax for the subroutine.
```

Comment:
Remove this comment. The simplest way to avoid the issue is to declare all the subroutines before executing any other programs. Besides,this fix probably won't work if CCL is configured to compile with optimize.